README.md

```diff
@@ -19,21 +19,31 @@ You can choose how much Clippy is supposed to ~~annoy~~ help you by changing the
 | `clippy::complexity`  | code that does something simple but in a complex way                                | **warn**      |
 | `clippy::perf`        | code that can be written to run faster                                              | **warn**      |
 | `clippy::pedantic`    | lints which are rather strict or have occasional false positives                    | allow         |
+| `clippy::restriction` | lints which prevent the use of language and library features[^restrict]             | allow         |
 | `clippy::nursery`     | new lints that are still under development                                          | allow         |
 | `clippy::cargo`       | lints for the cargo manifest                                                        | allow         |
 
 More to come, please [file an issue](https://github.com/rust-lang/rust-clippy/issues) if you have ideas!
 
-The [lint list](https://rust-lang.github.io/rust-clippy/master/index.html) also contains "restriction lints", which are
-for things which are usually not considered "bad", but may be useful to turn on in specific cases. These should be used
-very selectively, if at all.
+The `restriction` category should, *emphatically*, not be enabled as a whole. The contained
+lints may lint against perfectly reasonable code, may not have an alternative suggestion,
+and may contradict any other lints (including other categories). Lints should be considered
+on a case-by-case basis before enabling.
+
+[^restrict]: Some use cases for `restriction` lints include:
+
+    * Strict coding styles.
+    * Additional restrictions on CI (e.g. `clippy::todo`).
+    * Preventing panicking in certain functions (e.g. `#[forbid(clippy::unwrap_used)]` on a module or function).
```

Comment:
Here I would probably add links to our lint list. I'd also suggest making the example format uniform:

```suggestion
    * Strict coding styles (e.g. [`clippy::else_if_without_else`])
    * Additional restrictions on CI (e.g. [`clippy::todo`]).
    * Preventing functions that can panic (e.g. [`clippy::unwrap_used`]).

[`clippy::else_if_without_else`]: https://rust-lang.github.io/rust-clippy/master/index.html#else_if_without_else
[`clippy::todo`]: https://rust-lang.github.io/rust-clippy/master/index.html#todo
[`clippy::unwrap_used`]: https://rust-lang.github.io/rust-clippy/master/index.html#unwrap_used
```

Reply:
Moved the note about activating a lint only for a subset of the code to it's own item. I imagine it's not a thing people think of so might help to mention the possibility. Otherwise it's same as you suggested.

Reply:
Using a footnote seems a little out of place, but I think that's mostly because it's the only footnote.